game/scripts/vscripts/components/gold/hero_kills.lua

```diff
@@ -95,6 +95,24 @@ function HeroKillGold:HeroDeathHandler (keys)
 
   local killerEntity = keys.killer
   local killedHero = keys.killed
+  -- killer is sometimes nil for some reason
+  if not killerEntity then
+    local killedHeroName
+    if killedHero then
+      killedHeroName = killedHero:GetName()
+    else
+      killedHeroName = "Killed entity also nil ??????"
+    end
+    D2CustomLogging:sendPayloadForTracking(D2CustomLogging.LOG_LEVEL_INFO, "HERO DEATH EVENT FIRED WITH NIL KILLER", {
+      ErrorMessage = killedHeroName,
+      ErrorTime = GetSystemDate() .. " " .. GetSystemTime(),
+      GameVersion = GAME_VERSION,
+      DedicatedServers = (IsDedicatedServer() and 1) or 0,
+      MatchID = tostring(GameRules:GetMatchID())
+    })
+
+    return
+  end
   local killerTeam = killerEntity:GetTeamNumber()
   local killedTeam = killedHero:GetTeamNumber()
```

Comment:
The function just returns early when killerEntity is nil, since I have no idea what condition even causes that and the function can't really do its job when killer is unknown.